File: URI/cpp/2222.cpp
#include <iostream>
#include <vector>
#include <set>
#include <algorithm>    // std::set_intersection, std::sort
#include <iterator>
#include <bitset>





using namespace std;

int intersection(vector<int> a, vector<int> b){
  set <int, greater <int> > res;

  if( a.size() > b.size()){
    for(const int& i : a ){
      for(int k = 0; k < b.size();k++){
        if( b[k] == i){
          res.insert(i);
        }
      }
    }
  }else{
  for(const int& i : b ){
      for(int k = 0; k < a.size();k++){
        if( a[k] == i){
          res.insert(i);
        }
    }
  }
  }
  return res.size();
}

int union_(vector<int> a, vector<int> b){
  set <int, greater <int> > res;
  for(const int& i : a ){
    res.insert(i);
	}
  for(const int&l: b) {
    res.insert(l);
  }
  return res.size();
}
int main() {

    //reading an integer
      int a = 0;
      cin >> a; 
      for(int i = 0;i < a;i++) {
        int b,c,d,f,g,h;
        int e,s,k,j ;
        vector<bitset<61>> m;


        cin >>b;
        for(k =0;k < b;k++) {
          bitset<61> p;
          cin >> c;
          for(j=0; j < c;j++){
            cin >> d;
            p[d] = 1;
          }
          m.push_back(p);
        }
        cin >> e;
      for(s=0; s < e;s++){
        cin >> f;
        cin >> g;
        cin >> h;
        if( f == 1){
          cout << (m[g-1] & m[h-1]).count() << endl;
        }else{
          cout << (m[g-1] | m[h-1]).count() << endl;
        }
      }
  }
}

File: URI/go/2222.go
package main

import (
"fmt"
)
func intersection(a []uint32, b[]uint32)int{
  res := make(map[uint32]uint32)
  if len(a) > len(b){
    for _, n := range a {
      for t,_ := range b{
        if b[t] == n {
          res[n] = 1
        }
      }
    }
  }else{
    for _, n := range b {
      for t,_ := range a{
        if a[t] == n {
          res[n] = 1
        }
      }
    }
  }
  return len(res)
}

func union(a []uint32, b[]uint32)int{
  res := make(map[uint32]uint32)
  for _, n := range a {
    res[n] = 1
	}
  for _, n := range b {
    res[n] = 1
  }
  return len(res)
}

func main() {

    //reading an integer
      var a = 0
      fmt.Scan(&a)
      i := 0
      for i < a {
        var b,c,d,f,g,h uint32
        var e,s,k,j uint32
        m := [][]uint32{}


        fmt.Scan(&b)
        k = 0
        for k < b {
          r := []uint32{}
          fmt.Scan(&c)
          j = 0
          for j < c{
            fmt.Scan(&d)
            r = append(r, d)
            j++
          }
          m = append(m, r)
          k++
        }
      fmt.Scan(&e)
      s = 0
      for s < e{
        fmt.Scan(&f)
        fmt.Scan(&g)
        fmt.Scan(&h)
        if f == 1{
          fmt.Println(intersection(m[g-1], m[h-1]))
        }else{
          fmt.Println(union(m[g-1], m[h-1]))
        }
        s++
      }
      i++
  }
}
